src/client.ts

```diff
@@ -600,6 +616,140 @@ export default class ConcordiumNodeClient {
         }
     }
 
+    async getRewardStatus(
+        blockHash: string
+    ): Promise<RewardStatus | undefined> {
+        if (!isValidHash(blockHash)) {
+            throw new Error('The input was not a valid hash: ' + blockHash);
+        }
+
+        type DateKey = KeysMatching<RewardStatusV1, Date>;
+        type BigIntKey = KeysMatching<RewardStatusV0 & RewardStatusV1, bigint>;
+
+        const dates: DateKey[] = ['nextPaydayTime'];
+        const bigInts: BigIntKey[] = [
+            'protocolVersion',
+            'gasAccount',
+            'totalAmount',
+            'totalStakedCapital',
+            'bakingRewardAccount',
+            'totalEncryptedAmount',
+            'finalizationRewardAccount',
+            'foundationTransactionRewards',
+        ];
+
+        const bh = new BlockHash();
+        bh.setBlockHash(blockHash);
+
+        const response = await this.sendRequest(
+            this.client.getRewardStatus,
+            bh
+        );
+
+        return unwrapJsonResponse<RewardStatus>(
+            response,
+            buildJsonResponseReviver(dates, bigInts),
+            intToStringTransformer(bigInts)
+        );
+    }
+
+    /**
+     * Retrieve list of bakers on the network.
+     * @param blockHash the block hash to get the smart contact instances at
+     * @returns A JSON list of baker IDs
+     */
+    async getBakerList(blockHash: string): Promise<BakerId[] | undefined> {
+        if (!isValidHash(blockHash)) {
+            throw new Error('The input was not a valid hash: ' + blockHash);
+        }
+
+        const bh = new BlockHash();
+        bh.setBlockHash(blockHash);
+
+        const response = await this.sendRequest(this.client.getBakerList, bh);
+
+        return unwrapJsonResponse<BakerId[]>(
+            response,
+            (_, v) => BigInt(v as string),
+            intListToStringList
+        );
+    }
+
+    /**
+     * Gets the status the L-pool.
+     * @param blockHash the block hash the status at
+     * @returns The status of the L-pool.
+     */
+    async getPoolStatus(blockHash: string): Promise<LPoolStatus | undefined>;
+    /**
+     * Gets the status a baker.
+     * @param blockHash the block hash the status at
+     * @param bakerId the ID of the baker to get the status for.
+     * @returns The status of the corresponding baker pool.
+     */
+    async getPoolStatus(
```

Comment:
If I don't provide a bakerId, I get a BakerPool type back, with bakerId 0.
When I do provide a bakerId, I get the LPool?

Reply:
Yes.. I experience this as well; might be an issue with the grpc function.